taskManager/src/main/java/com/kunmi/taskManager/taskManager/services/ProjectServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+        }
+
+        List<ProjectRecord> projectRecords = projects.stream()
+                .map(project -> new ProjectRecord(project.getId().toString(), project.getName())
+                ).toList();
+
+        return new ProjectListResponseRecord(projectRecords, null);
+    }
+
+    @Override
+    public GetProjectResponseRecord find(String email, UUID id) {
+        User currentUser = getCurrentUser(email);
+        Project project = projectRepository.findById(id).orElseThrow(() ->
+                new ProjectNotFoundException("Project not found for user: " + currentUser.getEmail()));
+
+        if (!project.getUser().getId().equals(currentUser.getId())) {
+            throw new ProjectNotFoundException("Project not found for user:" + currentUser.getEmail());
+        }
+
+        return new GetProjectResponseRecord(project.getId(), project.getName());
+    }
+
+    @Override
+    @Transactional
+    public ProjectResponseRecord create(ProjectRequestRecord projectRequestRecord, String email) {
+        User user = getCurrentUser(email);
+
+        Project project = new Project(projectRequestRecord.projectName());
+        project.setUser(user);
+        projectRepository.save(project);
+
+        log.info("Project: {} created for user {}", project.getName(), email);
+        return new ProjectResponseRecord(project.getId(), project.getName());
+    }
+
+    @Override
+    @Transactional
+    public UpdateProjectResponseRecord update(UUID id, UpdateProjectRequestRecord updateProjectRequestRecord, String email) {
+        User currentUser = getCurrentUser(email);
+        Project project = projectRepository.findById(updateProjectRequestRecord.id())
+                .orElseThrow(() -> new ProjectNotFoundException("Project not found for: " + currentUser.getEmail()));
+
+        if (id.equals(currentUser.getId())) {
+            throw new ProjectNotFoundException("Project not found for user: " + currentUser.getEmail());
+        }
+
+        project.setName(updateProjectRequestRecord.projectName());
+        projectRepository.save(project);
+        log.info("Project with ID {} updated successfully", project.getId());
+        return new UpdateProjectResponseRecord("Project updated successfully");
+    }
+
+    @Override
+    @Transactional
+    public DeleteProjectResponseRecord delete(String email, UUID id) {
+        User currentUser = getCurrentUser(email);
+        Project project = projectRepository.findById(id)
+                .orElseThrow(() -> new ProjectNotFoundException("Project not found for user: " + currentUser.getEmail()));
+
+        if (!project.getUser().getId().equals(currentUser.getId())) {
+            throw new ProjectNotFoundException(currentUser.getEmail() + " Unauthorized to delete this project");
+        }
+
+        projectRepository.deleteById(project.getId());
+        log.info("Project ID {} is successfully deleted", project.getId());
+        return new DeleteProjectResponseRecord("Project deleted successfully");
```

Comment:
In all methods text messages dont necessary. We use rest api here, so we can use http codes for this. Nobody will read this messages.